fdbserver/workloads/DataDistributionMetrics.actor.cpp

```diff
@@ -26,52 +26,119 @@
 
 struct DataDistributionMetricsWorkload : KVWorkload {
 
-	int numTransactions;
-	int writesPerTransaction;
-	int transactionsCommitted;
-	int numShards;
+	int numShards, readPerTx, writePerTx;
 	int64_t avgBytes;
+	double testDuration;
+	std::string keyPrefix;
+	PerfIntCounter commits, errors;
 
 	DataDistributionMetricsWorkload(WorkloadContext const& wcx)
-	  : KVWorkload(wcx), transactionsCommitted(0), numShards(0), avgBytes(0) {
-		numTransactions = getOption(options, LiteralStringRef("numTransactions"), 100);
-		writesPerTransaction = getOption(options, LiteralStringRef("writesPerTransaction"), 1000);
+	  : KVWorkload(wcx), numShards(0), avgBytes(0), commits("Commits"), errors("Errors") {
+		testDuration = getOption(options, LiteralStringRef("testDuration"), 10.0);
+		keyPrefix = getOption(options, LiteralStringRef("keyPrefix"), LiteralStringRef("DDMetrics")).toString();
+		readPerTx = getOption(options, LiteralStringRef("readPerTransaction"), 1);
+		writePerTx = getOption(options, LiteralStringRef("writePerTransaction"), 5 * readPerTx);
+		ASSERT(nodeCount > 1);
 	}
 
 	static Value getRandomValue() {
 		return Standalone<StringRef>(format("Value/%08d", deterministicRandom()->randomInt(0, 10e6)));
 	}
 
-	ACTOR static Future<Void> _start(Database cx, DataDistributionMetricsWorkload* self) {
-		state int tNum;
-		for (tNum = 0; tNum < self->numTransactions; ++tNum) {
-			loop {
-				state ReadYourWritesTransaction tr(cx);
-				try {
-					state int i;
-					for (i = 0; i < self->writesPerTransaction; ++i) {
-						tr.set(StringRef(format("Key/%08d", tNum * self->writesPerTransaction + i)), getRandomValue());
+	Key keyForIndex(int n) { return doubleToTestKey((double)n / nodeCount, keyPrefix); }
+
+	ACTOR static Future<Void> ddRWClient(Database cx, DataDistributionMetricsWorkload* self) {
+		loop {
+			state ReadYourWritesTransaction tr(cx);
+			state int i;
+			try {
+				for (i = 0; i < self->readPerTx; ++i)
+					wait(success(
+					    tr.get(self->keyForIndex(deterministicRandom()->randomInt(0, self->nodeCount))))); // read
+				for (i = 0; i < self->writePerTx; ++i)
+					tr.set(self->keyForIndex(deterministicRandom()->randomInt(0, self->nodeCount)),
+					       getRandomValue()); // write
+				wait(tr.commit());
+				++self->commits;
+			} catch (Error& e) {
+				wait(tr.onError(e));
+			}
+			tr.reset();
```

Comment:
This reset the backoff wait time for the transaction.

Can we remove this reset and move the loop wrap the try-catch?

Reply:
Yeah, I did not realize reset will influence the metrics. I will remove it.
The loop right now is containing the `try-catch` or I misunderstand your point?

Reply:
You understood it right. Just remove the reset. :)